File: Core.DomainModel/ItProject/Goal.cs
using System;

namespace Core.DomainModel.ItProject
{
    /// <summary>
    /// It project goal.
    /// </summary>
    public class Goal : Entity, IProjectModule, ISupportsUserSpecificAccessControl
    {
        /// <summary>
        /// Human readable identifier.
        /// </summary>
        /// <remarks>
        /// Called "brugervendt nøgle" in OIO.
        /// </remarks>
        /// <value>
        /// The human readable identifier.
        /// </value>
        public string HumanReadableId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Note { get; set; }
        public bool Measurable { get; set; }
        public TrafficLight Status { get; set; }

        public int? GoalTypeId { get; set; }
        public virtual GoalType GoalType { get; set; }

        public int GoalStatusId { get; set; }
        public virtual GoalStatus GoalStatus { get; set; }

        public DateTime? SubGoalDate1 { get; set; }
        public DateTime? SubGoalDate2 { get; set; }
        public DateTime? SubGoalDate3 { get; set; }

        public string SubGoal1 { get; set; }
        public string SubGoal2 { get; set; }
        public string SubGoal3 { get; set; }

        public string SubGoalRea1 { get; set; }
        public string SubGoalRea2 { get; set; }
        public string SubGoalRea3 { get; set; }

        public bool HasUserWriteAccess(User user)
        {
            return GoalStatus != null && GoalStatus.HasUserWriteAccess(user);
        }
    }
}

File: Presentation.Web/app/models/it-project/goal.ts
module Kitos.Models.ItProject {
	/** It project goal. */
    export interface IGoal extends IEntity {
        /** Human readable identifier. */
        HumanReadableId: string;
        Name: string;
        Description: string;
        Note: string;
        Measurable: boolean;
        Status: any;
        GoalTypeId: number;
        GoalType: IGoalType;
        GoalStatusId: number;
        GoalStatus: IGoalStatus;
        SubGoalDate1: Date;
        SubGoalDate2: Date;
        SubGoalDate3: Date;
        SubGoal1: string;
        SubGoal2: string;
        SubGoal3: string;
        SubGoalRea1: string;
        SubGoalRea2: string;
        SubGoalRea3: string;
    }
}
